src/particles.cpp

```diff
@@ -302,128 +354,33 @@ void ParticleSpawner::step(float dtime, ClientEnvironment* env)
 		}
 	}
 
-	if (m_spawntime != 0) // Spawner exists for a predefined timespan
-	{
-		for(std::vector<float>::iterator i = m_spawntimes.begin();
-				i != m_spawntimes.end();)
-		{
-			if ((*i) <= m_time && m_amount > 0)
-			{
+	if (m_spawntime != 0) {
+		// Spawner exists for a predefined timespan
+		for (std::vector<float>::iterator i = m_spawntimes.begin();
```

Comment:
please use c++11 styled loop ? :)

Reply:
Can't, as we erase entries from the vector while iterating.